labrad/protocol.py

```
@@ -203,8 +205,20 @@ def packetReceived(self, source, context, request, records):
         else:
             self.messageReceived(source, context, records)
 
+    @inlineCallbacks
     def requestReceived(self, source, context, request, records):
         """Process incoming request."""
+        try:
+            if self.request_handler is None:
+                log.msg('server request_handler not set')
+                raise Exception('server request_handler not set')
+            response = yield self.request_handler(source, context, records)
+            self.sendPacket(source, context, -request, response)
+        except Exception as e:
+            # this will only happen if there was a problem while sending,
+            # which usually means a problem flattening the response into
+            # a valid LabRAD packet
+            self.sendPacket(source, context, -request, [(0, e)])
```

Doesn't this swallow the 'server request_handler not set' error?